hw/application_fpga/fw/tk1/lib.c

```diff
@@ -38,17 +38,15 @@ static void htif_set_tohost(uint8_t dev, uint8_t cmd, int64_t data)
 	htif_send(dev, cmd, data);
 }
 
-static int htif_putchar(int ch)
+static void htif_putchar(char ch)
 {
-	htif_set_tohost(1, 1, ch & 0xff);
-	return ch & 0xff;
+	htif_set_tohost((uint8_t)1, (uint8_t)1, (int64_t)ch & 0xff);
```

Comment:
Just to check: ch is defined in function header to be char. And then cast to int64_t. Do we send eight bytes?

Reply:
I'm not really familiar with the RISC-V Host to Target Interface (HTIF) console but according to our code there is a 64 bit union with data for both receving and sending. Check the `struct tohost`, `struct fromhost` and the union inside the function `htif_send()`.

We only use this in the firmware for debug prints when compiled without `-DNOCONSOLE`.  This means that when running on real hardware we don't use this at all.

TKey program don't use this interface at all, not even for qemu debug prints.